src/data/complex/circle.lean

```diff
@@ -0,0 +1,112 @@
+/-
+Copyright (c) 2021 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+import geometry.manifold.instances.sphere
+
+/-!
+# The circle
+
+This file defines `circle` to be the metric sphere (`metric.sphere`) in `ℂ` centred at `0` of
+radius `1`.  We equip it with the following structure:
+
+* a submonoid of `ℂ`
+* a group
+* a topological group
+* a charted space with model space `euclidean_space ℝ (fin 1)` (inherited from `metric.sphere`)
+* a Lie group with model space `𝓡 1`
```

Comment:
```suggestion
* a Lie group with model with corners `𝓡 1`
```
(The previous formulation seems to imply that there are two different model spaces involved, which is not the case)